### PR TITLE
[feature/cumulative--10.0.0] AB#33500 - Select Component 

### PR DESCRIPTION
**Pull request by fantaJinMode**

### CODE REVIEW

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        onFocus,
+        onSelect,
+        option,
+    }) => (
+        <div
+            className="Select-option"
+            key={key}
+            onClick={() => onSelect(option)}
+            onFocus={() => {}}
+            onKeyDown={() => {}}
```
Comment (groberts314) [inline docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx:27]: Let's use `noop` from Lodash.
I know this is just an example, but we want our examples to look like how we want actual code to be written! 😁

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                clearable
+                creatable
+                id="block--creatable_advanced_select_id"
+                options={options}
+                optionComponent={formatOptionLabelCustom}
+                onChange={onSelectOption}
+                placeholder="Select Option"
+                promptTextCreator={formatCreatableLabelOption}
+                searchable
+                selection
```
Comment (groberts314) [inline docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx:82]: I don't think we still need `selection` prop, right?  That's a holdover from `<Dropdown>`...?

File: docs/src/inputs/select/examples/exampleCreatableSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        <div>
+            <Select
+                clearable
+                creatable
+                id="block--creatable_select_id"
+                options={options}
+                onChange={onSelect}
+                placeholder="Select Option"
+                searchable
+                selection
```
Comment (groberts314) [inline docs/src/inputs/select/examples/exampleCreatableSelect.jsx:25]: Same here: I don't think we need `selection` prop.  Please remove wherever it is not needed.

**Review by groberts314 — CHANGES_REQUESTED** [review]
Looks pretty good to me.  Just a couple of minor things.

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx (hunk 1 above)
Comment (fantaJinMode) [reply docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx]: Fixed

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx (hunk 2 above)
Comment (fantaJinMode) [reply docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx]: @groberts314 Thank you for catching it. I have corrected it.

File: docs/src/inputs/select/examples/exampleCreatableSelect.jsx (hunk 3 above)
Comment (fantaJinMode) [reply docs/src/inputs/select/examples/exampleCreatableSelect.jsx]: @groberts314 Thank you for catching it. I have corrected it.

**Review by cbrewer1689 — APPROVED** [review]
👍

**Comment by cbrewer1689** [comment]
@sonurauniyar, please merge the cumulative branch into this one.

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        onFocus,
+        onSelect,
+        option,
+    }) => (
+        <div
+            className="Select-option"
+            key={key}
+            onClick={() => onSelect(option)}
+            onFocus={noop()}
+            onKeyDown={noop()}
```
Comment (groberts314) [inline docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx:30]: We need to remove the parentheses.
```JavaScript
onFocus={noop}
onKeyDown={noop}
```

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx (hunk 1 above)
Comment (groberts314) [reply docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx]: These anonymous noop lambdas were replaced by Lodash `noop` but there is a new issue which I [commented on separately](https://github.com/saddlebackdev/react-cm-ui/pull/247#discussion_r532816497).

Resolving.

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx (hunk 2 above)
Comment (groberts314) [reply docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx]: Yep, gonezo.  Resolving.

File: docs/src/inputs/select/examples/exampleCreatableSelect.jsx (hunk 3 above)
Comment (groberts314) [reply docs/src/inputs/select/examples/exampleCreatableSelect.jsx]: Gonezo; resolving.

File: docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx (hunk 4 above)
Comment (fantaJinMode) [reply docs/src/inputs/select/examples/exampleCreatableAdvancedSelect.jsx:30]: @groberts314 Fixed

**Review by groberts314 — APPROVED** [review]
Thanks for all the fixes @fantaJinMode 
Approved! :shipit:

**Pull request merged**